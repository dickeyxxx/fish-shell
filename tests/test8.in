# Test index ranges 

# Test variable expand
set test (seq 10)
echo $test[1..10]      # normal range
echo $test[10..1]      # inverted range
echo $test[2..5 8..6]  # several ranges
echo $test[-1..-2]     # range with negative limits
echo $test[-1..1]      # range with mixed limits

# Test variable set
set test1 $test
set test1[-1..1] $test  # reverse variable
echo $echo $test1
set test1[2..4 -2..-4] $test1[4..2 -4..-2]
echo $test1

# Test command substitution
echo (seq 5)[-1..1]
echo (seq 10)[3..5 -2..2]

