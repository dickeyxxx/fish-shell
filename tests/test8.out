1 2 3 4 5 6 7 8 9 10
10 9 8 7 6 5 4 3 2 1
2 3 4 5 8 7 6
10 9
10 9 8 7 6 5 4 3 2 1
10 9 8 7 6 5 4 3 2 1
10 7 8 9 6 5 2 3 4 1
5 4 3 2 1
3 4 5 9 8 7 6 5 4 3 2
